### DISCUSSION TITLE
MySQL Workbench called LearningSQL

### DISCUSSION THREAD

**Discussion by Faithsakho** [community/community#76142] [Programming Help]
Answer the following questions using Joins in SQL. Please note that there might be several different ways to join the tables (e.g., inner join, where, nested query, etc.).

1. Create a list of all employees first and last names, and the name of the branch. Use  INNER JOIN ON syntax. 
2. Using an inner join, show the number of employees in each department, grouped by department. 
3. Using a one-sided outer join, show the number of employees in each department.
4. Show a list of the total balance for each Branch, including the branch name (using inner join).
5. Retrieve a list of employees who worked with individual customers (i.e., CUST_TYPE_CD = ‘I’). Display those employees’ last name, first name, title, and respective CustomerID.  (Hint: joining account, employee, and customer)

**Comment by kur0neko** [comment]
1.  List of all employees first and last names, and the name of the branch. Use INNER JOIN ON syntax.

```sql
SELECT e.first_name, e.last_name, b.branch_name
FROM employees e
INNER JOIN employee_department ed ON e.employee_id = ed.employee_id
INNER JOIN departments d ON ed.department_id = d.department_id
INNER JOIN branches b ON d.branch_id = b.branch_id;
```
2.  Using an inner join, show the number of employees in each department, grouped by department.

```sql 
SELECT d.department_name, COUNT(e.employee_id) AS employee_count
FROM departments d
INNER JOIN employee_department ed ON d.department_id = ed.department_id
INNER JOIN employees e ON ed.employee_id = e.employee_id
GROUP BY d.department_name;

```
3. Using a one-sided outer join, show the number of employees in each department.
```sql
SELECT d.department_name, COUNT(e.employee_id) AS employee_count
FROM departments d
LEFT JOIN employee_department ed ON d.department_id = ed.department_id
LEFT JOIN employees e ON ed.employee_id = e.employee_id
GROUP BY d.department_name;
```
4. Show a list of the total balance for each Branch, including the branch name (using inner join).

```sql
SELECT b.branch_name, SUM(a.balance) AS total_balance
FROM branches b
INNER JOIN accounts a ON b.branch_id = a.branch_id
GROUP BY b.branch_name;
```
5. Retrieve a list of employees who worked with individual customers (i.e., CUST_TYPE_CD = 'I'). Display those employees’ last name, first name, title, and respective CustomerID. (Hint: joining account, employee, and customer)

```sql
SELECT e.last_name, e.first_name, e.title, c.customer_id
FROM employees e
INNER JOIN accounts a ON e.employee_id = a.employee_id
INNER JOIN customers c ON a.customer_id = c.customer_id
WHERE c.CUST_TYPE_CD = 'I';
```
These are all examples
